modality-probe-capi/modality-probe-capi-impl/src/lib.rs

```diff
@@ -199,29 +196,55 @@ fn distribute_error_to_modality_probe_error(
 /// to an initialized instance operating in a single-threaded
 /// fashion.
 #[cfg_attr(feature = "no_mangle", no_mangle)]
-pub unsafe fn modality_probe_distribute_snapshot(
+pub unsafe fn modality_probe_produce_snapshot(
     probe: *mut ModalityProbe<'static>,
     destination_snapshot: *mut CausalSnapshot,
 ) -> ModalityProbeError {
     let probe = match probe.as_mut() {
         Some(t) => t,
         None => return MODALITY_PROBE_ERROR_NULL_POINTER,
     };
-    match probe.distribute_snapshot() {
+    match probe.produce_snapshot() {
         Ok(snapshot) => {
             *destination_snapshot = snapshot;
             MODALITY_PROBE_ERROR_OK
         }
-        Err(e) => distribute_error_to_modality_probe_error(e),
+        Err(e) => produce_error_to_modality_probe_error(e),
+    }
+}
+
+/// # Safety
+///
+/// The ModalityProbe instance pointer must be non-null and point
+/// to an initialized instance operating in a single-threaded
+/// fashion.
+#[cfg_attr(feature = "no_mangle", no_mangle)]
+pub unsafe fn modality_probe_produce_snapshot_bytes(
+    probe: *mut ModalityProbe<'static>,
+    history_destination: *mut u8,
+    history_destination_bytes: usize,
+    out_written_bytes: *mut usize,
+) -> ModalityProbeError {
+    let probe = match probe.as_mut() {
+        Some(t) => t,
+        None => return MODALITY_PROBE_ERROR_NULL_POINTER,
+    };
+    match probe.produce_snapshot_bytes(core::slice::from_raw_parts_mut(
+        history_destination,
```

Comment:
Should we also null-check the destination pointer?

Reply:
:+1: 228c86cb03e059e38c0e19ffc480bd5884a1fe35